### PR TITLE
PERF-4347 Combine QE workloads using conditional sharding

### PR DESCRIPTION
**Pull request by alicedoherty**
**Jira Ticket:** [PERF-4347](https://jira.mongodb.org/browse/PERF-4347)

**Whats Changed:**  
- The non-sharded and sharded versions of the [YCSB-like queryable encryption](https://github.com/mongodb/genny/tree/master/src/workloads/encrypted) workloads have been combined into one file using the [OnlyRunInInstance functionality](https://github.com/mongodb/genny/blob/master/src/cast_core/src/RunCommand.cpp#L216).
- (Note: I've also updated [ExponentialCompact.yml](https://github.com/mongodb/genny/blob/master/src/workloads/encrypted/ExponentialCompact.yml) even though it doesn't have a sharded version because it referenced the `ShardCollectionPhase` param that I cleaned up to use `OnlyRunInInstance: sharded` instead).

**Patch Testing Results:**  
- [Updated QE workloads running on Linux-1-Node ReplSet FLE 2022-11](https://spruce.mongodb.com/version/65295b5f3627e0465ba94e59/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) and [Perf Analyser results](https://performance-analyzer.server-tig.prod.corp.mongodb.com/perf-analyzer-viz/?comparison_id=e41e6b67-768b-4fd4-8db9-e46fd2616e17&percent_filter=0%7C%7C100&z_filter=0%7C%7C10)
- [Updated QE workloads running on Linux Shard Lite FLE 2022-11](https://spruce.mongodb.com/version/6526b8c81e2d1765dc75ab91/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC) (can't use Perf Analyser to look at the sharded tests as the task name has changed - but manual inspection showed no performance changes).

**Related PRs:**   
- [SERVER-82178](https://jira.mongodb.org/browse/SERVER-82178) and https://github.com/10gen/mongo/pull/16239

### CODE REVIEW

**Comment by roganamy** [comment]
See comment on https://github.com/10gen/mongo/pull/16239/files
You'll need another reviewer for both PRs aswell

**Comment by roganamy** [comment]
Do we expect to see a performance difference with these changes?

File: src/phases/encrypted/ContinuousWritesWithExponentialCompactTemplate.yml [hunk 1]
```diff
@@ -85,7 +68,20 @@ Actors:
...
+        Operations:
+        - OperationMetricsName: EnableSharding
+          OperationName: AdminCommand
+          OperationCommand:
+            enableSharding: *encrypted_db
+        - OperationMetricsName: ShardCollection
+          OperationName: AdminCommand
+          OperationCommand:
+            shardCollection: *encrypted_ns
+            key: {field9: hashed}
```
Comment (erwee) [inline src/phases/encrypted/ContinuousWritesWithExponentialCompactTemplate.yml:84]:
```suggestion
            key: {field1: hashed}
```

**Review by erwee — CHANGES_REQUESTED** [review]
Glad to see our workload files reduced by a half. Thanks for doing this work!

File: src/phases/encrypted/ContinuousWritesWithExponentialCompactTemplate.yml (hunk 1 above)
Comment (alicedoherty) [reply src/phases/encrypted/ContinuousWritesWithExponentialCompactTemplate.yml]: Copy and paste error sorry! Nice catch.

**Review by erwee — APPROVED** [review]

**Review by roganamy — APPROVED** [review]

**Comment by alicedoherty** [comment]
evergreen merge

**Pull request merged**